extensions/sql-database-projects/src/models/project.ts

```diff
@@ -45,23 +44,117 @@ export class Project {
 
 		// find all folders and files to include
 
-		for (const itemGroup of result['ItemGroup']) {
-			if (itemGroup['Build'] !== undefined) {
-				for (const fileEntry of itemGroup['Build']) {
-					this.files.push(this.createProjectEntry(fileEntry.$['Include'], EntryType.File));
-				}
+		for (let ig = 0; ig < this.projFileXmlDoc.documentElement.getElementsByTagName('ItemGroup').length; ig++) {
+			const itemGroup = this.projFileXmlDoc.documentElement.getElementsByTagName('ItemGroup')[ig];
+
+			for (let b = 0; b < itemGroup.getElementsByTagName('Build').length; b++) {
+				this.files.push(this.createProjectEntry(itemGroup.getElementsByTagName('Build')[b].getAttribute('Include'), EntryType.File));
 			}
 
-			if (itemGroup['Folder'] !== undefined) {
-				for (const folderEntry of itemGroup['Folder']) {
-					this.files.push(this.createProjectEntry(folderEntry.$['Include'], EntryType.Folder));
-				}
+			for (let f = 0; f < itemGroup.getElementsByTagName('Folder').length; f++) {
```

Comment:
add to constants

Reply:
same thing for all strings here

---
In reply to: [406337045](https://github.com/microsoft/azuredatastudio/pull/9884#discussion_r406337045) [](ancestors = 406337045)